pkg/drawio/nodesRelations.go

```diff
@@ -0,0 +1,193 @@
+		}
+		for _, tn := range append(info.dstGroupingLines, info.srcGroupingLines...) {
+			data.svgNames[tn] = data.svgNames[line]
+		}
+		data.svgNames[info.srcGroupingPoint] = data.svgNames[line]
+		data.svgNames[info.dstGroupingPoint] = data.svgNames[line]
+	}
+}
+
+// nodeParents() - return the parents - basically, if node is presented, all its parents are presented
+func nodeParents(node TreeNodeInterface) []TreeNodeInterface {
+	if node == nil {
+		return nil
+	}
+	return append(nodeParents(node.Parent()), node)
+}
+
+// nodeSubTree() - return the subtree - basically, all the subtree of a square is presented with the square
+func nodeSubTree(node TreeNodeInterface) []TreeNodeInterface {
+	nodes := getAllSquares(node)
+	for _, icon := range getAllIcons(node) {
+		if !icon.(IconTreeNodeInterface).IsGroupingPoint() {
+			nodes = append(nodes, icon)
+		}
+	}
+	return nodes
+}
+
+// sgTreeNodes() calc the SG tn that are part of the SG (spatial case of getSubtree() )
+// SG can have in more than one  case
+func sgTreeNodes(sgTn *SGTreeNode) (psg, icons []TreeNodeInterface) {
+	for _, psgTn := range sgTn.partialSgs {
+		psg = append(psg, psgTn)
+	}
+	for _, icon := range sgTn.elements {
+		icons = append(icons, icon)
+	}
+	return psg, icons
+}
+
+// lineInfo is a struct with all the treeNodes representing a grouping line:
+// for example, for a connection  A,B -> C,D wi will can have the following on the canvas:
+// A->gp1, B->gp1, gp1->gp2, gp2->C, gp2->D
+// we have a groupSquares for [A,B] and [C,D], these are hold as src and dst.
+// (notice that A,B,C,D are not stored here)
+type lineInfo struct {
+	mainLine                           LineTreeNodeInterface
+	src, dst                           TreeNodeInterface
+	router                             TreeNodeInterface
+	srcGroupingLines, dstGroupingLines []LineTreeNodeInterface
+	srcGroupingPoint, dstGroupingPoint TreeNodeInterface
+}
+
+// getLineInfo() calc all the treeNodes representing a grouping line (or just a line) and return it in a lineInfo struct
+func getLineInfo(line LineTreeNodeInterface) *lineInfo {
+	src := line.Src()
+	dst := line.Dst()
+	info := &lineInfo{line, src, dst, line.Router(), nil, nil, nil, nil}
+
+	srcIsGP := src.IsIcon() && src.(IconTreeNodeInterface).IsGroupingPoint()
+	dstIsGP := dst.IsIcon() && dst.(IconTreeNodeInterface).IsGroupingPoint()
+
+	if srcIsGP && slices.Contains(src.(*GroupPointTreeNode).groupedIconsConns, line) ||
+		dstIsGP && slices.Contains(dst.(*GroupPointTreeNode).groupedIconsConns, line) {
+		// this is the case that the line is one of the A->gp1, B->gp1, gp2->C, gp2->D
+		return nil
+	}
+	if srcIsGP {
+		info.src = src.Parent()
+		info.srcGroupingLines = src.(*GroupPointTreeNode).groupedIconsConns
+		info.srcGroupingPoint = src
+	}
+	if dstIsGP {
+		info.dst = dst.Parent()
+		info.dstGroupingLines = dst.(*GroupPointTreeNode).groupedIconsConns
+		info.dstGroupingPoint = dst
+	}
+	// if both src and dst are not grouping point, its just a regular line
```

Comment:
This comment confused me a little bit. I'm not sure it is needed.

Reply:
comment elaborated